### PR TITLE
better highlights

### PR DESCRIPTION
**Pull request by Decodetalkers**
I add some keyword to get better highlight

### CODE REVIEW

**Comment by Decodetalkers** [comment]
image
know it looks like this kind.. however, it is sitll white in the block

**Comment by Decodetalkers** [comment]
Can you help provide better highlight? I am not very familiar with the highlight of treesitter..

**Comment by yuja** [comment]
> Can you help provide better highlight? I am not very familiar with the highlight of treesitter..

I'm not familiar either, but blocks are basically javascript so I think you can copy rules from the neovim javascript mode.

With `tree-sitter highlight` command (which loads qml/javascript/typescript highlight rules in order), code blocks are highlighted as follows:
image

File: queries/highlights.scm [hunk 1]
```diff
@@ -44,7 +44,18 @@
 (ui_signal_parameter
   (identifier) @variable.parameter)
 
+;;; ui_object_definition
+(ui_object_definition
+  type_name: (identifier) @type)
```
Comment (yuja) [inline queries/highlights.scm:49]: This one looks good.

The other rules are redundant since they are included in highlights-javascript.scm.

**Review by yuja — CHANGES_REQUESTED** [review]

File: queries/highlights.scm (hunk 1 above)
Comment (Decodetalkers) [reply queries/highlights.scm:49]: Thanks. .. but how to include scm in scheme?

Comment (Decodetalkers) [reply queries/highlights.scm:49]: and I have problem in the command of tree-sitter hightlight, how to get the highlight?
image
I failed..

Comment (yuja) [reply queries/highlights.scm:49]: I think that's up to neovim and I don't know.

For example, emacs-tree-sitter has its own highlight rules. So I think there would be a neovim way.
https://github.com/emacs-tree-sitter/tree-sitter-langs/tree/master/queries

Regarding the tree-sitter CLI, .scm files are specified in package.json.

Comment (Decodetalkers) [reply queries/highlights.scm:49]: > I think that's up to neovim and I don't know.
> 
> For example, emacs-tree-sitter has its own highlight rules. So I think there would be a neovim way. https://github.com/emacs-tree-sitter/tree-sitter-langs/tree/master/queries
> 
> Regarding the tree-sitter CLI, .scm files are specified in package.json.

I mean , I only have the highlights.scm, and if I want to include the scm of javascript , must I copy the text of javascript or I can just use "inculde" to include the scheme?

Comment (yuja) [reply queries/highlights.scm:49]: > I mean , I only have the highlights.scm as the entrance, and if I want to include the scm of javascript , must I copy the text of javascript or I can just use "inculde" to include the scheme?

AFAIK, there's no "include" stuff. just copy the content or load multiple .scm files.

> and I have problem in the command of tree-sitter hightlight, how to get the highlight?

https://tree-sitter.github.io/tree-sitter/syntax-highlighting#paths

You need to add tree-sitter-qmljs directory to "parser-directories".

My ~/.config/tree-sitter/config.json:
```
{
  "parser-directories": [
    "/home/yuya/work/2021/tree-sitter",
    "/home/yuya/github",
    "/home/yuya/src",
    "/home/yuya/source"
  ],
  "theme": {
    "attribute": {
      "color": 124,
      "italic": true
    },
    "constant": 94,
    "module": 136,
    "keyword": 56,
    "property": 124,
    "operator": {
      "color": 239,
      "bold": true
    },
    "punctuation.bracket": 239,
    "embedded": null,
    "function.builtin": {
      "bold": true,
      "color": 26
    },
    "variable.parameter": {
      "underline": true
    },
    "number": {
      "color": 94,
      "bold": true
    },
    "variable.builtin": {
      "bold": true
    },
    "function": 26,
    "constant.builtin": {
      "bold": true,
      "color": 94
    },
    "string": 28,
    "constructor": 136,
    "punctuation.delimiter": 239,
    "tag": 18,
    "type": 23,
    "comment": {
      "color": 245,
      "italic": true
    },
    "string.special": 30,
    "type.builtin": {
      "bold": true,
      "color": 23
    }
  }
}

```

**Pull request closed without merging**